cmd: Verifying container image if policy controller is present. (#2703)
inspektor-gadget/inspektor-gadget: docs/getting-started/verify.md

## Verify the container image while deploying

> When deploying Inspektor Gadget using `kubectl gadget deploy`, the image will be automatically verified if the
> `policy-controller` is already present in your kubernetes cluster.

Reviewer: ```suggestion
When deploying Inspektor Gadget using `kubectl gadget deploy`, the image will be automatically verified if the
`policy-controller` is deployed on your Kubernetes cluster.
```

ghcr.io/inspektor-gadget/inspektor-gadget@sha256:9272c2be979a9857971fc8b6f7226e609cadec8352f97e9769081930121ef27f signature key validation failed for authority authority-0 for ghcr.io/inspektor-gadget/inspektor-gadget@sha256:9272c2be979a9857971fc8b6f7226e609cadec8352f97e9769081930121ef27f: no matching signatures
```

As this image is not signed, the verification failed and the container was not deployed to the cluster.

In case the `policy-controller` is not present, a warning message will be printed to inform you the verification will not take place:

```bash
$ kubectl get pod -n cosign-system
No resources found in cosign-system namespace.
$ kubectl gadget deploy
WARN[0000] No policy controller found, the container image will not be verified
...
Inspektor Gadget successfully deployed
```

### Skipping verification

You can also decide to not verify the image, using `--verify-image=false`.
We definitely do not recommend this and you will use this option at your own risk:

```bash
$ kubectl gadget deploy --verify-image=false
WARN[0000] You used --verify-image=false, the container image will not be verified
...
Inspektor Gadget successfully deployed
```

> ### Using custom public key for verification

Reviewer: Do you have any idea why this line isn't rendered properly?
